src/CMakeLists.txt

```diff
@@ -56,18 +56,41 @@ find_package(Threads REQUIRED)
 set(Boost_USE_MULTITHREADED ON)
 
 add_library(_espressopp ${ESPRESSO_SOURCES})
+
+# compiler specific configurations
+if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
+    if(ESPP_LOCAL_ARCHITECTURE)
+        target_compile_options(_espressopp PRIVATE "-march=native")
```

Comment:
Will it lead to problems if only the library is compiled with optimization flags, but not the executables (tests)?

Reply:
No, this does not change the ABI. It only tells the compiler which instruction set it is allowed to used.